From 5a26d306bfc6f427b4ddf1a36500d11c9980f90a Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Fri, 8 Jan 2016 13:36:05 +0800
Subject: [PATCH 1599/5242] MLK-12161 tty: serial: imx: only enable RTSD
 interrupt for hw flow control

commit  865cb241395e476b6f29131cb224b7198799ad02 from
https://source.codeaurora.org/external/imx/linux-imx.git

Only enable RTSD interrupt for hw flow control, otherwise RTS_B signal
has some external signal disturbance without config RTS_B select input.

Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit: 471e8c43aca3)

Conflicts:
	drivers/tty/serial/imx.c
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 905b0dc..fe66880 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1422,7 +1422,9 @@ static int imx_uart_startup(struct uart_port *port)
 	ucr1 = imx_uart_readl(sport, UCR1);
 	if (!sport->dma_is_inited)
 		ucr1 |= UCR1_RRDYEN;
-	ucr1 |= UCR1_RTSDEN | UCR1_UARTEN;
+	if (sport->have_rtscts)
+		ucr1 |= UCR1_RTSDEN;
+	ucr1 |= UCR1_UARTEN;
 	imx_uart_writel(sport, ucr1, UCR1);
 
 	ucr4 = imx_uart_readl(sport, UCR4) & ~UCR4_OREN;
-- 
1.7.9.5

